### PR TITLE
feat: TypeScript example of passing resources between stacks

### PR DESCRIPTION
**Pull request by richardhboyd**
This demonstrates how to pass resources between stacks in TypeScript.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

**Comment by richardhboyd** [comment]
> Title does not follow the guidelines of [Conventional Commits](https://www.conventionalcommits.org). Please adjust title before merge.

@mergify you got some nerve after your shenanigans this week.

File: typescript/cross-stack-resources/bin/cross-stack-resources.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env node
+import cdk = require('@aws-cdk/core');
+import { ApplicationStack } from '../lib/application-stack';
+import { InfrastructureStack } from '../lib/infrastructure-stack';
+
+const app = new cdk.App();
+const infra = new InfrastructureStack(app, "InfrastructureStack");
```
Comment (NGL321) [inline typescript/cross-stack-resources/bin/cross-stack-resources.ts]: Where is this stack?

File: typescript/cross-stack-resources/bin/cross-stack-resources.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+// We can now reference the myMainLambda property of the infra object.
+
+const myApp = new ApplicationStack(app, 'ApplicationStack', {
```
Comment (NGL321) [inline typescript/cross-stack-resources/bin/cross-stack-resources.ts]: ^=== See above

**Comment by NGL321** [comment]
Closing ftm. Can reopen when there is time to finish

😸

**Pull request closed without merging**